pkg/didcomm/protocol/introduce/service_test.go

```diff
@@ -32,2640 +32,1293 @@ import (
-	didEvent                chan<- service.StateMsg
-	controller              *gomock.Controller
-	messenger               service.Messenger
-	startWithRequest        bool
-	startWithProposal       bool
-	skipProposal            bool
-	withResponseError       bool
-	withSecondResponseError bool
-	withSecondResponseStop  bool
-	withProposalStop        bool
-	withRequestStop         bool
-	withResponseStop        bool
+// payload represents a transport message structure
+type payload struct {
+	msg      []byte
+	myDID    string
+	theirDID string
 }
 
-func TestService_New(t *testing.T) {
-	const errMsg = "test err"
-
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	t.Run("OpenStore Error", func(t *testing.T) {
-		storageProvider := storageMocks.NewMockProvider(ctrl)
-		storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(nil, errors.New(errMsg))
-
-		provider := introduceMocks.NewMockProvider(ctrl)
-		provider.EXPECT().StorageProvider().Return(storageProvider)
-
-		svc, err := introduce.New(provider)
-		require.EqualError(t, err, "test err")
-		require.Nil(t, svc)
-	})
-
-	t.Run("Service Error", func(t *testing.T) {
-		storageProvider := storageMocks.NewMockProvider(ctrl)
-		storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(nil, nil)
-
-		provider := introduceMocks.NewMockProvider(ctrl)
-		provider.EXPECT().StorageProvider().Return(storageProvider)
-		provider.EXPECT().Service(didexchange.DIDExchange).Return(nil, errors.New(errMsg))
-
-		svc, err := introduce.New(provider)
-		require.EqualError(t, err, "load the DIDExchange service: test err")
-		require.Nil(t, svc)
-	})
-
-	t.Run("Cast Service Error", func(t *testing.T) {
-		storageProvider := storageMocks.NewMockProvider(ctrl)
-		storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(nil, nil)
-
-		provider := introduceMocks.NewMockProvider(ctrl)
-		provider.EXPECT().StorageProvider().Return(storageProvider)
-		provider.EXPECT().Service(didexchange.DIDExchange).Return(nil, nil)
-
-		svc, err := introduce.New(provider)
-		require.EqualError(t, err, "cast service to service.Event")
-		require.Nil(t, svc)
-	})
-}
-
-func TestService_Stop(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	storageProvider := storageMocks.NewMockProvider(ctrl)
-	storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(nil, nil)
-
-	didService := serviceMocks.NewMockDIDComm(ctrl)
-	didService.EXPECT().RegisterMsgEvent(gomock.Any()).Return(nil)
-	didService.EXPECT().UnregisterMsgEvent(gomock.Any()).Return(nil)
-
-	provider := introduceMocks.NewMockProvider(ctrl)
-	provider.EXPECT().StorageProvider().Return(storageProvider)
-	provider.EXPECT().Messenger().Return(nil)
-	provider.EXPECT().Service(didexchange.DIDExchange).Return(didService, nil)
-
-	svc, err := introduce.New(provider)
-	require.NoError(t, err)
-
-	require.NoError(t, svc.Stop())
-	require.EqualError(t, svc.Stop(), "server was already stopped")
-}
-
-func TestService_Name(t *testing.T) {
```

Comment:
i see the tests are mainly deleted. are they not needed anymore?

Reply:
tests I have deleted are useless. Some functions were removed from the service. e.g Stop.